src/routers/course.ts

```diff
@@ -32,41 +26,52 @@ courseRouter.post('/', async (req, res) => {
 	try {
 		const userID = req.header('userID');
 		if (userID) {
-			const course = new Course({
-				owner: userID,
-				name: req.body.name,
-			});
-			const created = await create(course);
-			res.json(created);
+			const body = req.body;
+			const checkPost = ajv.compile(
+				schema.components.schemas.CourseCreate
+			);
```

Comment:
You might be able to define these outside of the functions above where you instantiated AJV